core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.ts

```diff
@@ -0,0 +1,193 @@
+    this._clonedFrom = clonedFrom;
+    this._isCommunityOwned = isCommunityOwned;
+    this._viewableIfPrivate = viewableIfPrivate;
+  }
+
+  clonedFrom(): string {
+    return this._clonedFrom;
+  }
+
+  isCloned(): boolean {
+    return Boolean(this._clonedFrom);
+  }
+  isPrivate(): boolean {
+    return this._status === constants.ACTIVITY_STATUS_PRIVATE;
+  }
+  isPublic(): boolean {
+    return this._status === constants.ACTIVITY_STATUS_PUBLIC;
+  }
+  isCommunityOwned(): boolean {
+    return this._isCommunityOwned;
+  }
+  viewableIfPrivate(): boolean {
+    return this._viewableIfPrivate;
+  }
+  makeCommunityOwned(): Promise<void> {
+    let that = this;
+    let requestUrl = (
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      make_community_owned: true
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+
+  setViewability(viewableIfPrivate: boolean): Promise<void> {
+    let that = this;
+    let requestUrl = (
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      viewable_if_private: viewableIfPrivate
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+  saveRoleChanges(
+      newMemberUsername: string, newMemberRole: string): Promise<void> {
+    let that = this;
+    let requestUrl = (
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      new_member_role: newMemberRole,
+      new_member_username: newMemberUsername
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      that.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+  publish(): Promise<void> {
```

Comment:
For all the functions that return a promise, Find its usages and inside the .then() put `$rootScope.$applyAsync()` if it is in angularjs.

Reply:
done!

Reply:
@srijanreddy98 PTAL